admin-dev/themes/new-theme/js/pages/order/create/order-product-component.js

```diff
@@ -0,0 +1,313 @@
+/**
+ * 2007-2019 PrestaShop SA and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import createOrderPageMap from '../create-order-map';
+
+const $ = window.$;
+
+/**
+ * Page Object for "Create order" page
+ */
+export default class OrderProductComponent {
+  constructor() {
+    this.products = [];
+    this.combinations = [];
+
+    this._initEvents();
+
+    return {};
+  }
+
+  /**
+   * Initialize page's events.
+   *
+   * @private
+   */
+  _initEvents() {
+    $(createOrderPageMap.productSearch).on('input', (event) => this._handleProductSearch(event));
+    $(createOrderPageMap.productSelect).on('change', (event) => this._handleProductChange(event));
+  }
+
+  /**
+   * Searches for product
+   *
+   * @private
+   */
+  _handleProductSearch(event) {
+    const name = $(event.target).val();
+
+    if (name.length < 3) {
+      return;
+    }
+
+    $.ajax($(event.target).data('url'), {
+      method: 'GET',
+      data: {
+        product_search_phrase: name,
+      },
+    }).then((response) => {
+      this.products = JSON.parse(response);
+      this._renderProductSearchResult();
```

Comment:
i would as well but because i need products data in other function it needs to be set globally here or in _renderProductSearchResult method. do you think setting in _renderProductSearchResult is better?

Reply:
Ohh, then it doesn't matter I guess.